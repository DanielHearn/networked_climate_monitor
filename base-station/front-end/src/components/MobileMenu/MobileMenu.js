export default {
  name: 'mobile-menu',
  props: {
    loggedIn: {
      type: Boolean,
      default: false,
      required: true
    }
  },
  methods: {
    toggleMenu: function() {
      this.$store.commit('setMobileMenu', !this.$store.state.mobileMenu)
    }
  }
}
