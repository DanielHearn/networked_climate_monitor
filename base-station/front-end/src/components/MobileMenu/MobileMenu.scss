@import "./../../scss/variables.scss";

.mobile-menu {
  display: flex;
  flex: 1 auto;
  background: $darkest;
  justify-content: center;
  align-items: center;
  ul {
    display: flex;
    flex-direction: column;
    padding: 0;
    .link {
      color: $white;
      padding: 1em 0;
      font-size: $m-size;
      font-weight: $medium-weight;
      text-decoration: none;
      &:hover {
        text-decoration: underline;
      }
    }
    .router-link-exact-active {
      text-decoration: underline;
    }
  }
}