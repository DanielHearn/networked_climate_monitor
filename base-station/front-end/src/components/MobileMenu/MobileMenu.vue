<template>
  <div class="mobile-menu">
    <ul @click="toggleMenu">
      <router-link to="/" class="link">Home</router-link>
      <template v-if="!loggedIn">
        <router-link to="/login" class="link">Login</router-link>
        <router-link to="/register" class="link">Register</router-link>
      </template>
      <template v-if="loggedIn">
        <router-link to="/dashboard" v-if="loggedIn" class="link"
          >Dashboard</router-link
        >
        <router-link to="/settings" v-if="loggedIn" class="link"
          >Settings</router-link
        >
      </template>
    </ul>
  </div>
</template>

<script src="./MobileMenu.js"></script>

<style lang="scss" src="./MobileMenu.scss"></style>
