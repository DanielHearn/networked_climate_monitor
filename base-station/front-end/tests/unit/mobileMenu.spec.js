import Vuex from 'vuex'
import { createLocalVue, shallowMount } from '@vue/test-utils'
import MobileMenu from './../../src/components/MobileMenu/MobileMenu.vue'

const localVue = createLocalVue()
localVue.use(Vuex)

const mutations = {
  setMobileMenu: jest.fn()
}

describe('MobileMenu.vue', () => {
  let wrapper
  let store

  beforeEach(() => {
    store = new Vuex.Store({ mutations })
    wrapper = shallowMount(MobileMenu, {
      propsData: {
        loggedIn: false
      },
      stubs: ['router-link'],
      store,
      localVue
    })
  })

  afterEach(() => {
    wrapper.destroy()
    jest.resetAllMocks()
  })

  it('Logged In', async () => {
    wrapper.setProps({ loggedIn: true })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('Mobile menu toggle', async () => {
    wrapper.find('ul').trigger('click')
    await wrapper.vm.$nextTick()
    expect(mutations.setMobileMenu).toHaveBeenCalledTimes(1)
  })

  it('has the expected html structure', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
