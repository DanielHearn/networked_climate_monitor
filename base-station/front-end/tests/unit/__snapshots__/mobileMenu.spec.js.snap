// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileMenu.vue Logged In 1`] = `
<div
  class="mobile-menu"
>
  <ul>
    <router-link-stub
      class="link"
      to="/"
    >
      Home
    </router-link-stub>
     
    <!---->
     
    <router-link-stub
      class="link"
      to="/dashboard"
    >
      Dashboard
    </router-link-stub>
     
    <router-link-stub
      class="link"
      to="/settings"
    >
      Settings
    </router-link-stub>
  </ul>
</div>
`;

exports[`MobileMenu.vue has the expected html structure 1`] = `
<div
  class="mobile-menu"
>
  <ul>
    <router-link-stub
      class="link"
      to="/"
    >
      Home
    </router-link-stub>
     
    <router-link-stub
      class="link"
      to="/login"
    >
      Login
    </router-link-stub>
     
    <router-link-stub
      class="link"
      to="/register"
    >
      Register
    </router-link-stub>
     
    <!---->
  </ul>
</div>
`;
